### PR TITLE
Cknieling/problem view

### PR DESCRIPTION
**Pull request by chr11115**
- report cgaErrors
- improve UX of error inspection by adding a problem window

### CODE REVIEW

File: src/serlio/utils/MayaUtilities.cpp [hunk 1]
```diff
@@ -44,4 +44,14 @@ std::filesystem::path getWorkspaceRoot(MStatus& status) {
 		return {};
 	}
 }
+bool mStringArraysAreEqual(const MStringArray& lhs, const MStringArray& rhs) {
```
Comment (mistafunk) [inline src/serlio/utils/MayaUtilities.cpp:47]: consider implementing this as `operator==`, would be more versatile

File: src/serlio/utils/MayaUtilities.cpp [hunk 2]
```diff
@@ -44,4 +44,14 @@ std::filesystem::path getWorkspaceRoot(MStatus& status) {
...
+	if (lhs.length() != rhs.length())
+		return false;
+	uint32_t index;
```
Comment (mistafunk) [inline src/serlio/utils/MayaUtilities.cpp:50]: unused variable

**Review by mistafunk — CHANGES_REQUESTED** [review]
Looks mostly good, two tiny code issue.

One functional issue is that long messages are hard to see in the new window. Try to soft wrap the message or if not possible, to add scrollbars as a fallback.
image

File: src/serlio/utils/MayaUtilities.cpp (hunk 2 above)
Comment (chr11115) [reply src/serlio/utils/MayaUtilities.cpp]: fixed.

File: src/serlio/utils/MayaUtilities.cpp (hunk 1 above)
Comment (chr11115) [reply src/serlio/utils/MayaUtilities.cpp]: fixed.

**Review by mistafunk — CHANGES_REQUESTED** [review]
Code looks good now, unfortunately I found two functional issues:
* Once a problem view window is open, it is not refreshed once the RPK of the originating node changes -> please refresh the window once the RPK changes
* For multiple Serlio nodes and open problem views it is not clear which window belongs to which node -> please add the node name to the window title (remember to refresh it if the node name changes), also add the current RPK name (without path)

**Comment by chr11115** [comment]
Problem Views are now more dynamic:

-  The Title of the window now shows the referenced node and rule file
-  They automatically update on node rename
-  They automatically update on rule changes
-  They correctly handle multiple instances of the same rule
-  They automatically close, if the referenced node nolonger exists (i.e. is deleted)

**Comment by chr11115** [comment]
Added separate field to show rule (with entire path) instead of showing the rule name in the title:
image

File: src/serlio/modifiers/MayaCallbacks.cpp [hunk 3]
```diff
@@ -414,13 +414,18 @@ std::filesystem::path getAssetDir() {
...
-		const bool shouldBeLogged =
-		        (std::wcsstr(message, L"newer than current") || (level == prt::CGAErrorLevel::CGAERROR));
-
+	if (message != nullptr) {
+		bool shouldBeLogged = (level == prt::CGAErrorLevel::CGAERROR);
 		std::wstring stringMessage = message;
-		prtu::replaceCGACWithCEVersion(stringMessage);
-		cgacErrors.emplace_back(level, shouldBeLogged, stringMessage);
+
+		if (std::wcsstr(message, L"CGAC version")) {
```
Comment (mistafunk) [inline src/serlio/modifiers/MayaCallbacks.cpp:421]: Consider explicitly writing the ` != nullptr` check for `std::wcsstr`

Please update all wcsstr occurrences.

**Review by mistafunk — APPROVED** [review]
LGTM, very nice work!

Found one tiny issue regarding readbility of std::wcsstr, afterwards feel free to merge directly.

**Review by mistafunk — APPROVED** [review]
good stuff, ready to merge!

**Pull request merged**